Spigot-Server-Patches/0604-Stop-entities-loading-chunks-when-running-checks.patch

```diff
@@ -0,0 +1,299 @@
+                 this.setMot(vec3d2.x, vec3d2.y, 0.0D);
+             }
+ 
+-            Block block = iblockdata.getBlock();
++            Block block = null;
++            if (iblockdata != null) block = iblockdata.getBlock();
+ 
+-            if (vec3d.y != vec3d1.y) {
++            if (block != null && vec3d.y != vec3d1.y) {
+                 block.a((IBlockAccess) this.world, this);
+             }
++            // Paper end
+ 
+             // CraftBukkit start
+             if (positionChanged && getBukkitEntity() instanceof Vehicle) {
+@@ -673,11 +677,11 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
+             }
+             // CraftBukkit end
+ 
+-            if (this.onGround && !this.bv()) {
++            if (block != null && this.onGround && !this.bv()) { // Paper - Avoid loading chunks when performing movement checks
+                 block.stepOn(this.world, blockposition, this);
+             }
+ 
+-            if (this.playStepSound() && !this.isPassenger()) {
++            if (block != null && this.playStepSound() && !this.isPassenger()) { // Paper - Avoid loading chunks when performing movement checks
+                 double d0 = vec3d1.x;
+                 double d1 = vec3d1.y;
+                 double d2 = vec3d1.z;
+@@ -737,12 +741,19 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
+         }
+     }
+ 
+-    protected BlockPosition ap() {
++    // Paper start - Don't attempt to spawn sprint particles if flying
++    protected BlockPosition getBlockPositionUnder() {
+         int i = MathHelper.floor(this.loc.x);
+         int j = MathHelper.floor(this.loc.y - 0.20000000298023224D);
+         int k = MathHelper.floor(this.loc.z);
+         BlockPosition blockposition = new BlockPosition(i, j, k);
+ 
++        return blockposition;
++    }
++
++    protected BlockPosition ap() {
++        BlockPosition blockposition = this.getBlockPositionUnder();
++
+         if (this.world.getType(blockposition).isAir()) {
+             BlockPosition blockposition1 = blockposition.down();
+             IBlockData iblockdata = this.world.getType(blockposition1);
+@@ -755,6 +766,7 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
+ 
+         return blockposition;
+     }
++    // Paper end
+ 
+     protected float getBlockJumpFactor() {
+         float f = this.world.getType(this.getChunkCoordinates()).getBlock().getJumpFactor();
+@@ -972,17 +984,24 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
+                 for (int j = blockposition.getY(); j <= blockposition1.getY(); ++j) {
+                     for (int k = blockposition.getZ(); k <= blockposition1.getZ(); ++k) {
+                         blockposition_mutableblockposition.d(i, j, k);
+-                        IBlockData iblockdata = this.world.getType(blockposition_mutableblockposition);
+ 
+-                        try {
```

Comment:
Please avoid changing the indentation of the existing code here.

Reply:
I might be missing something here, but the indentation here is due to an added if statement. It looks like the comments are in the wrong spot however, which might be the actual issue here.

Reply:
min-diff, changing the indentation causes more diff noise than is needed, causing un-needed code conflicts when merging in upstream changes, etc, makes the cost of keeping it pretty too high to be worth it

Reply:
Ah I see what you mean.

Reply:
Believe latest commit should resolve this